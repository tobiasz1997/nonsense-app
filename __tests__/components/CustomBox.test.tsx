import { render, screen } from '@testing-library/react';
import CustomBox from '@components/ui/CustomBox';

describe('Custom Box', () => {
	const title = 'Test Title';
	const content = 'Test Content';
	it('should render without crashing', () => {
		const { container } = render(<CustomBox title={title}></CustomBox>);
		expect(container).toMatchSnapshot();
	});

	it('should title be correct', () => {
		render(<CustomBox title={title}></CustomBox>);

		const titleInDoc = screen.getByText(title);

		expect(titleInDoc).toBeInTheDocument();
	});

	it('should content be correct', () => {
		render(<CustomBox title={title}>{content}</CustomBox>);

		const contentInDoc = screen.getByText(content);

		expect(contentInDoc).toBeInTheDocument();
	});
});
