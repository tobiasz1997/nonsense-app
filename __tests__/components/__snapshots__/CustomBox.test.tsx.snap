// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Box should render without crashing 1`] = `
<div>
  <section
    class="space-y-5 rounded-xl bg-pistachio p-5 shadow-xl dark:bg-zinc-500"
  >
    <h2
      class="text-xl font-bold text-orange sm:text-3xl"
    >
      Test Title
    </h2>
  </section>
</div>
`;
